VK_App/VK_App/LoginViewController.swift

```diff
@@ -0,0 +1,121 @@
+// LoginViewController.swift
+// Copyright © RoadMap. All rights reserved.
+
+import UIKit
+
+/// Экран входа в приложение
+final class LoginViewController: UIViewController {
+    private enum Constants {
+        static let emailText = "1"
+        static let passwordText = "1234"
+        static let tabBarIdentifier = "tabBar"
+        static let alertTitleText = "Ошибка"
+        static let alertMessegeText = "Введены неверные данные пользователя"
+        static let alertActionTitleText = "OK"
+    }
+
+    // MARK: - Private IBOutlet
+
+    @IBOutlet private var scrollView: UIScrollView!
+    @IBOutlet private var emailTextField: UITextField!
+    @IBOutlet private var passwordTextField: UITextField!
+
+    // MARK: - Life Cycles
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupUI()
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        NotificationCenter.default.addObserver(
+            self,
+            selector: #selector(keyboardWasShown),
+            name: UIResponder.keyboardWillShowNotification,
+            object: nil
+        )
+
+        NotificationCenter.default.addObserver(
+            self,
+            selector: #selector(keyboardWillBeHidden),
+            name: UIResponder.keyboardWillHideNotification,
+            object: nil
+        )
+    }
+
+    override func viewWillDisappear(_ animated: Bool) {
```

Comment:
```suggestion
    override func viewWillDisappear(_ animated: Bool) {
```
Джан а где супер? Осуждаю

Reply:
согласен